src/SphereXProtectedBase.sol

```diff
@@ -4,327 +4,16 @@
 pragma solidity ^0.8.0;
 
 import {ISphereXEngine, ModifierLocals} from "./ISphereXEngine.sol";
+import {SphereXProtectedBaseUpgradeable} from "./SphereXProtectedBaseUpgradeable.sol";
 
 /**
  * @title SphereX base Customer contract template
  */
-abstract contract SphereXProtectedBase {
-    /**
-     * @dev we would like to avoid occupying storage slots
-     * @dev to easily incorporate with existing contracts
-     */
-    bytes32 private constant SPHEREX_ADMIN_STORAGE_SLOT = bytes32(uint256(keccak256("eip1967.spherex.spherex")) - 1);
-    bytes32 private constant SPHEREX_PENDING_ADMIN_STORAGE_SLOT =
-        bytes32(uint256(keccak256("eip1967.spherex.pending")) - 1);
-    bytes32 private constant SPHEREX_OPERATOR_STORAGE_SLOT = bytes32(uint256(keccak256("eip1967.spherex.operator")) - 1);
-    bytes32 private constant SPHEREX_ENGINE_STORAGE_SLOT =
-        bytes32(uint256(keccak256("eip1967.spherex.spherex_engine")) - 1);
-
-    event ChangedSpherexOperator(address oldSphereXAdmin, address newSphereXAdmin);
-    event ChangedSpherexEngineAddress(address oldEngineAddress, address newEngineAddress);
-    event SpherexAdminTransferStarted(address currentAdmin, address pendingAdmin);
-    event SpherexAdminTransferCompleted(address oldAdmin, address newAdmin);
-    event NewAllowedSenderOnchain(address sender);
-
+abstract contract SphereXProtectedBase is SphereXProtectedBaseUpgradeable {
```

Comment:
it is confusing that it inherits from upgradable but it is not upgradable? or it is? If we want to make it not upgradable you should add disable_init logic and call it from the constructor.

Reply:
are we planning to use this for non upgradable inline implementations and the BaseUpgradable for upgradavle iniline implementation?
or this will be for all inline and the baseupgradable was created just for the proxy create2 stuff?